### DISCUSSION TITLE
posts/why-we-prefer-property-over-field/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#14] [Announcements]
# posts/why-we-prefer-property-over-field/

为什么我们不直接把自动属性（也就是那种只写了“get;set;”的属性）直接写成公共字段？

https://blog.coldwind.top/posts/why-we-prefer-property-over-field/

**Comment by BYJRK** [comment]
写了文章再照着读一遍，做成一期视频，应该不算水吧？

**Reply by xiaososa** [reply]
不算！期待视频~我再看一遍加深记忆

**Comment by Captain-Tom-CL** [comment]
field这个语法糖可太甜了吧!

**Reply by BYJRK** [reply]
哈哈是的，本来以为 .NET 7 就有了，当时激动了半天，结果没来

**Comment by aubreychao** [comment]
如果属性A的set里用到了属性B的field那怎么写

**Reply by BYJRK** [reply]
写完整属性

**Comment by ZXX-9527-001** [comment]
还有一个重要原因，IDE会分析出属性的引用，方便项目维护。

**Reply by BYJRK** [reply]
这个字段也可以的，只是IDE（比如VS）中默认不会像属性那样在代码中显示，而是需要手动查看引用
